### DISCUSSION TITLE
ComplexHeatmap labels

### DISCUSSION THREAD

**Discussion by brianpenghe** [GreenleafLab/ArchR#1138] [Questions / Documentation]
I was trying to plot motifs using these:
```
heatmapEM <- plotEnrichHeatmap(enrichMotifs, n = 7, transpose = TRUE)
options(repr.plot.width = 20, repr.plot.height = 10)
ComplexHeatmap::draw(heatmapEM, heatmap_legend_side = "bot", annotation_legend_side = "bot")

```
But the labels on top are so far away from each other. Anyway to fix it?
![image](https://user-images.githubusercontent.com/4110443/140088043-22279eb0-b480-4863-9fbf-ca40f52fdabd.png)

**Comment by rcorces** [comment]
This is a question about ComplexHeatmap, not ArchR. I'd recommend asking this in a general forum or on the ComplexHeatmap github

**Reply by brianpenghe** [reply]
OK I posted this https://github.com/jokergoo/ComplexHeatmap/issues/837#issue-1045290482  Hope people with similar questions can find it useful.

**Reply by brianpenghe** [reply]
> This is a question about ComplexHeatmap, not ArchR. I'd recommend asking this in a general forum or on the ComplexHeatmap github

It seems that the use of anno_link or anno_mark in .ArchRHeatmap is the reason? Is there an easy way for me to use anno_text instead?

**Reply by rcorces** [reply]
The plotting functions in ArchR are our best effort to make something that is flexible enough for most use cases. If you look at the code for `.ArchRHeatmap()` it is really just jumping through a lot of hoops to create a ComplexHeatmap object. You are welcome to create your own version of that function or build your own heatmap from scratch using the raw data.
